python/update_package_properties.py

```diff
@@ -0,0 +1,346 @@
+    return debian_latest_changelog
+
+
+def prepend_latest_changelog_into_debian_changelog(changelog_param: ChangelogParams, changelog_file_path: str) -> None:
+    debian_latest_changelog = convert_citus_changelog_into_debian_changelog(changelog_param)
+    with open(changelog_file_path, "r+") as reader:
+        if not (f"({changelog_param.get_project_version()}" in reader.readline()):
+            reader.seek(0, 0)
+            old_changelog = reader.read()
+            changelog = f"{debian_latest_changelog}{old_changelog}"
+            reader.seek(0, 0)
+            reader.write(changelog)
+        else:
+            raise ValueError("Already version in the debian changelog")
+
+
+@validate_parameters
+def update_pkgvars(project_name: str, version: is_version(non_empty(no_whitespaces(non_blank(str)))), fancy: bool,
+                   fancy_release_count: non_negative(int), templates_path: str, pkgvars_path: str) -> None:
+    env = get_template_environment(templates_path)
+
+    version_str = get_version_number_with_project_name(project_name,version, fancy, fancy_release_count)
+
+    template = env.get_template('pkgvars.tmpl')
+
+    pkgvars_content = f"{template.render(version=version_str)}\n"
+    with open(f'{pkgvars_path}/pkgvars', "w") as writer:
+        writer.write(pkgvars_content)
+
+
+def get_rpm_changelog_history(spec_file_path: str) -> str:
+    with open(spec_file_path, "r") as reader:
+        spec_content = reader.read()
+        changelog_index = spec_content.find("%changelog")
+        changelog_content = spec_content[changelog_index + len("%changelog") + 1:]
+
+    return changelog_content
+
+
+def get_rpm_header(changelog_params: ChangelogParams):
+    formatted_date = changelog_params.get_changelog_date().strftime("%a %b %d %Y")
+    return f"* {formatted_date} - {changelog_params.get_name_surname()} <{changelog_params.get_microsoft_email()}> " \
+           f"{get_version_number_with_project_name(changelog_params.get_project_name(), changelog_params.get_project_version(), changelog_params.get_fancy(), changelog_params.get_fancy_version_number())} "
+
+
+def get_debian_trailer(microsoft_email: str, name_surname: str, changelog_date: date):
+    formatted_date = changelog_date.strftime("%a, %d %b %Y %H:%M:%S %z ")
+    return f" -- {name_surname} <{microsoft_email}>  {formatted_date} \n "
+
+
+def convert_citus_changelog_into_rpm_changelog(changelog_params: ChangelogParams) -> str:
+    header = get_rpm_header(changelog_params)
+    rpm_changelog = f"{header.strip()}\n- Official {changelog_params.get_project_version()} release of {changelog_params.get_project_name().capitalize()}"
+
+    return rpm_changelog
+
+
+def update_rpm_spec(changelog_param: ChangelogParams, spec_file_name: str,
+                    templates_path: str) -> None:
+    env = get_template_environment(templates_path)
+
+    rpm_version = get_rpm_version(changelog_param.get_project_name(), changelog_param.get_project_version())
```

Comment:
By making this a method of `ChangelogParams` you can instead do:
```suggestion
def update_rpm_spec(self, spec_file_name: str,
                    templates_path: str) -> None:
    env = get_template_environment(templates_path)

    rpm_version = get_rpm_version(self.project_name, self.project_version)
```

Reply:
I think update_rpm_spec should not belong to ChanglogParam  I think se should keep ChangelogParam as Plain old object